sys/net/include/netdev.h

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netdev Network device driver interface
+ * @addtogroup  net
+ * @{
+ *
+ * @file
+ * @brief       Network device driver interface definition.
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef __NETDEV_H_
+#define __NETDEV_H_
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Type for @ref msg_t if device fired an event.
+ */
+#define NETDEV_MSG_EVENT_TYPE   (0x0100)
+
+/**
+ * @brief   Definition of basic network device options.
+ * @note    Feel free to expand if your device needs/supports more.
+ */
+typedef enum {
+    NETDEV_OPT_CHANNEL,             /**< Channel for the device as unsigned value
+                                         in host byte order */
+    NETDEV_OPT_IS_CHANNEL_CLR,      /**< Check if channel is clear */
+    NETDEV_OPT_ADDRESS,             /**< Hardware address for the device as
+                                         unsigned value in host byte order */
+    NETDEV_OPT_NID,                 /**< Network ID (e.g. PAN ID in IEEE 802.15.4)
+                                         for the device as unsigned value in
+                                         host byte order */
+    NETDEV_OPT_ADDRESS_LONG,        /**< Longer hardware address for the device
+                                         (e.g. EUI-64) for the device as
+                                         unsigned value in host byte order */
+    NETDEV_OPT_TX_POWER,            /**< The output of the device in dB as
+                                         signed value in host byte order */
+    NETDEV_OPT_MAX_PACKET_SIZE,     /**< Maximum packet size the device supports
+                                         unsigned value in host byte order */
+    NETDEV_OPT_SRC_LEN,             /**< Default mode the source address is
+                                         set to as value of `size_t`. (e.g.
+                                         either PAN-centric 16-bit address or
+                                         EUI-64 in IEEE 802.15.4) */
+    NETDEV_OPT_EN_PRELOADING,       /**< Enable pre-loading of data, transfer
```

Comment:
Please document what should be expected for `set_option()` parameters `data` and `data_len`, and that these are only allowed for `set_option()`, but not `get_option()`.